tests/test_sync_types.py

```diff
@@ -0,0 +1,142 @@
+from fastavro import writer, reader, parse_schema
+import os
+from datetime import date, time, datetime, timezone
+from uuid import uuid4
+from decimal import Decimal
+import decimal
+from datetime import date
+from datetime import datetime
+from datetime import time
+from decimal import Decimal
+from typing import TypedDict, Optional
+from uuid import uuid4
+import snapshottest
+
+
+nested_record_schema = {
+    "namespace": "example.avro",
+    "type": "record",
+    "name": "User",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "favorite_number", "type": ["null", "int"]},
+        {"name": "favorite_color", "type": ["null", "string"]},
+        {"name": "birthdate", "type": {"type": "int", "logicalType": "date"}},
+        {"name": "appt_date", "type": {"type": "int", "logicalType": "date"}},
+        {
+            "name": "time_of_day_birth",
+            "type": {"type": "int", "logicalType": "time-millis"},
+        },
+        {
+            "name": "timestamp_of_birth",
+            "type": {"type": "long", "logicalType": "timestamp-millis"},
+        },
+        {
+            "name": "uuid_of_birth_record",
+            "type": {"type": "string", "logicalType": "uuid"},
+        },
+        {
+            "name": "weight",
+            "type": {
+                "type": "bytes",
+                "logicalType": "decimal",
+                "precision": 5,
+                "scale": 2,
+            },
+        },
+        {
+            "name": "address",
+            "type": {
+                "type": "record",
+                "name": "AddressUSRecord",
+                "fields": [
+                    {"name": "streetaddress", "type": "string"},
+                    {"name": "city", "type": "string"},
+                ],
+            },
+        },
+    ],
+}
+
+
+class ExampleAvroAddressUSRecord(TypedDict):
+    streetaddress: str
+    city: str
+
+
+class ExampleAvroUser(TypedDict):
+    name: str
+    favorite_number: Optional[int]
+    favorite_color: Optional[str]
+    birthdate: date
+    appt_date: date
+    time_of_day_birth: time
+    timestamp_of_birth: datetime
+    uuid_of_birth_record: uuid4
+    weight: Decimal
+    address: ExampleAvroAddressUSRecord
+
+
+class SnapshotTestSyncedTypes(snapshottest.TestCase):
+    def test_synced_types(self):
+        self.maxDiff = None
+        parsed_schema = parse_schema(nested_record_schema)
+        records = [
```

Comment:
Can we add type hints here
```suggestion
        records: list[ExampleAvroUser] = [
```

Reply:
done

Reply:
done